app/Models/Foundation/Summit/Registration/PromoCodes/SummitRegistrationPromoCode.php

```diff
@@ -728,4 +728,21 @@ public function addTicket(SummitAttendeeTicket $ticket): void
         $ticket->setPromoCode($this);
         $this->tickets->add($ticket);
     }
+
+    /**
+     * @param SummitTicketType $ticket_type
+     * @return SummitAttendeeTicket|null
+     */
+    public function getNextAvailableTicketPerType(SummitTicketType $ticket_type): ?SummitAttendeeTicket
+    {
+        $criteria = Criteria::create();
```

Comment:
run this conde on a do while bc un need to lock temporally the ticket on cache using a TTL ( 5 minutes or so)
so if u get next available one, u need to check that is not already locked by a previous attempt 
so once u got a ticket u need to add it to a lock key on cache